src/Delivery/DynamicEntry.php

```diff
@@ -88,51 +94,73 @@ public function __call($name, $arguments)
             $getId = true;
         }
 
-        if ($fieldConfig !== null && !$fieldConfig->isDisabled()) {
-            if (!isset($this->fields->$fieldName)) {
-                if ($fieldConfig->getType() === 'Array') {
-                    return [];
-                }
+        if ($fieldConfig === null || $fieldConfig->isDisabled()) {
+            trigger_error('Call to undefined method ' . __CLASS__ . '::' . $name . '()', E_USER_ERROR);
+        }
 
-                return null;
+        if (!isset($this->fields->$fieldName)) {
+            if ($fieldConfig->getType() === 'Array') {
+                return [];
             }
 
-            $value = $this->fields->$fieldName;
-            if (!$fieldConfig->isLocalized()) {
-                $locale = $this->getSpace()->getDefaultLocale()->getCode();
-            }
+            return null;
+        }
 
-            $result = $value->$locale;
-            if ($getId && !($fieldConfig->getType() === 'Link' || ($fieldConfig->getType() === 'Array' && $fieldConfig->getItemsType() === 'Link'))) {
-                trigger_error('Call to undefined method ' . __CLASS__ . '::' . $name . '()', E_USER_ERROR);
-            }
+        if ($getId && !($fieldConfig->getType() === 'Link' || ($fieldConfig->getType() === 'Array' && $fieldConfig->getItemsType() === 'Link'))) {
+            trigger_error('Call to undefined method ' . __CLASS__ . '::' . $name . '()', E_USER_ERROR);
+        }
 
-            if ($getId && $fieldConfig->getType() === 'Link') {
-                return $result->getId();
+        $value = $this->fields->$fieldName;
+        if (!$fieldConfig->isLocalized()) {
+            $locale = $this->getSpace()->getDefaultLocale()->getCode();
+        } else {
+            while (!isset($value->$locale)) {
```

Comment:
Do you want to think about an endless loop protection? A counter or something, that you don't spin for ever, if the input data is _invalid_?